### PR TITLE
[catboost/java] fixing the NativeLib on jdk 9+

### PR DESCRIPTION
**Pull request by ixtora**
Fix for issue #852

### CODE REVIEW

File: catboost/jvm-packages/catboost4j-prediction/pom.xml [hunk 1]
```diff
@@ -107,6 +107,7 @@
                     <encoding>
                         utf-8
                     </encoding>
+                    <source>8</source>
```
Comment (qdeee) [inline catboost/jvm-packages/catboost4j-prediction/pom.xml]: Why have you added this line?

Comment (ixtora) [reply catboost/jvm-packages/catboost4j-prediction/pom.xml]: https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8212233 -
(maven + javadoc + jdk 11-12) the problem is not solved.

**Comment by annaveronika** [comment]
@arcadia-devtools Ship it!

**Comment by arcadia-devtools** [comment]
@ixtora, please agree to the CLA (https://github.com/catboost/catboost/blob/master/CONTRIBUTING.md#provide-contributions)

**Comment by arcadia-devtools** [comment]
@annaveronika, internal review request created: [836798](https://a.yandex-team.ru/review/836798)

**Comment by ixtora** [comment]
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

**Comment by arcadia-devtools** [comment]
Check [836798](https://a.yandex-team.ru/review/836798) passed

**Comment by annaveronika** [comment]
@ixtora Thanks a lot! We're merging the pr.

**Pull request merged**